### DISCUSSION TITLE
Test dice scores using ensemble of 5 folds better than Validation dice scores from individual validation

### DISCUSSION THREAD

**Discussion by rbramkumar** [MIC-DKFZ/nnUNet#2773] [Q&A]
Just wanted to ask the community about a situation that I am seeing with my models. 

When using 5-fold cross validation, the dice scores on each CV set is around **_0.45-0.55_** dice score. However when I use the ensemble of all 5 folds to predict on the Test (hold-out) set, my overall mean dice on test set is around **_0.60._** 

I am thinking it may either be the case that the ensemble is better than each individual model, or that the the test set is somehow more representative of the training cases where the model is able to learn well. 

Has anyone else noticed the first scenario i.e., even with random or stratified selection of test/hold-out set, the test accuracy is better than individual CV set. How are you all reporting these scores? I am guessing if we show lower CV scores and higher Test scores, there needs to be more explanation about the test cases selection, and maybe explanation on how the ensemble scores perform better?

Thank you in advance!

**Comment by rbramkumar** [comment]
I'm not sure if anyone has a better suggestion but I'm going the route of further splitting the training set to carve out another validation set. It's not optimal since it's reducing the training cases further, but it's needed to validate the ensembles prior to test. 

So, previously I was only splitting my dataset of 41 patients into 33 train and 8 test. The CV split for each fold within 33 patients was 28, 5. Now, I've carved out another validation set of 5 patients from the training set of 33. So my CV splits are from 28 train cases i.e., 24, 4. 

The ensemble from nnUNET is compared with other algorithms on the new carved out validation set of 5 patients. 

If there any other best practices other than carving out another validation dataset from an already small dataset, it would be helpful.
